docs/strongly-typed-reducers.md

```diff
@@ -0,0 +1,166 @@
+# Strongly Typed Reducers
+
+It's good practice in typescript to be as specific about your types as possible.
+This helps you catch errors at compile-time instead of run-time.
+
+Reducers are no exception to this rule. However it's not always obvious how to
+make this happen in practice.
+
+## Reducer Typing Best Practices
+
+### Define an Interface for your State
+
+It's important to strongly type the data in your store, and this is done by
+defining types for the `state` arguments to your reducers:
+
+```typescript
+export type TFoo: string;
+
+// Being explicit about the state argument and return types ensures that all your
+// reducer's cases return the correct type. 
+export const fooReducer = (state: TFoo, action): TFoo => {
+  // ...
+};
+
+export interface IBar {
+  a: number;
+  b: string;
+}
+
+export const barReducer = (state: IBar, action): IBar => {
+  // ...
+};
+```
+
+Since most applications are composed of several reducers, you should compose 
+a global 'AppState' by composing the reducer types:
+
+```typescript
+export interface IAppState {
```

Comment:
What about:
```js
export interface IAppState {
   foo?: TFoo;
   bar?: IBar;
 }
```

So you're able to provide an empty object when configuring the store. I pretty much do it this way all the time.